kie-pmml/src/main/java/org/kie/pmml/pmml_4_2/PMML4ExecutionHelper.java

```
@@ -0,0 +1,342 @@
+                KieBaseConfiguration kieBaseConf) {
+            return new PMML4ExecutionHelper(modelName,content,kieBaseConf,false);
+        }
+
+        public static PMML4ExecutionHelper getExecutionHelper(String modelName,
+                byte[] content,
+                KieBaseConfiguration kieBaseConf,
+                boolean includeMiningDataSources) {
+            return new PMML4ExecutionHelper(modelName,content,kieBaseConf,includeMiningDataSources);
+        }
+
+        public static PMML4ExecutionHelper getExecutionHelper(String modelName,
+                Resource resource,
+                KieBaseConfiguration kieBaseConf) {
+            return new PMML4ExecutionHelper(modelName,resource,kieBaseConf,false);
+        }
+
+        public static PMML4ExecutionHelper getExecutionHelper(String modelName,
+                Resource resource,
+                KieBaseConfiguration kieBaseConf,
+                boolean includeMiningDataSources) {
+            return new PMML4ExecutionHelper(modelName,resource,kieBaseConf,includeMiningDataSources);
+        }
+    }
+
+    private PMML4ExecutionHelper(String modelName, KieBase kbase, boolean includeMiningDataSources) {
+        this.modelName = modelName;
+        this.possiblePackageNames = new ArrayList<>();
+        this.kbase = kbase;
+        this.includeMiningDataSources = includeMiningDataSources;
+        initRuleUnitExecutor();
+    }
+
+
+
+    private PMML4ExecutionHelper(String modelName, String classpathName, KieBaseConfiguration kieBaseConf, boolean includeMiningDataSources) {
+        this.modelName = modelName;
+        this.possiblePackageNames = new ArrayList<>();
+        this.includeMiningDataSources = includeMiningDataSources;
+        try {
+            kbase = new KieHelper().addFromClassPath(classpathName).build(kieBaseConf);
+            initRuleUnitExecutor();
+        } catch (Exception e) {
+//            logger.error("Unable to initialize PMML4ExecutionHelper", e);
```

Swallowed exception. Also later in this class.